Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
```

Comment:
I'm fully with @bwaidelich aggregate-relevant metadata -> domain property this change make sense and avoid common trap where user start to ask where did I need to store this or that.